tests/test_evaluation.py

```
@@ -467,10 +466,30 @@ def test_evaluator_metrics(minimal_instance):
+
+def test_evaluator_logging_empty_frame_pairs(capsys, minimal_instance):
+    """Test that the Evaluator logs an error when there are no matching frame pairs."""
+    
+    logger.remove()
+    logger.add(sys.stderr, level="ERROR")
+    # Create user_labels and pred_labels that will lead to empty frame pairs
+    user_labels, pred_labels = create_labels_no_match_frame_pairs(minimal_instance)
+
+    # Use capsys to capture output
+    with capsys.disabled():  # Disable capturing to see print statements if needed
+        with pytest.raises(Exception):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)